api/filter/accesslog.proto

```diff
@@ -9,24 +9,53 @@ import "google/protobuf/struct.proto";
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/wrappers.proto";
 
-message AccessLogEntry {
-  // The HTTP request method (RFC 7231/2616)
-  RequestMethod request_method = 1;
+// Envoy access logs describe incoming interaction with Envoy over a fixed
+// period of time, and typically cover a single request/response exchange.
+// Except where explicitly declared otherwise, all fields describe
+// *downstream* interaction between Envoy and a connected client.
+// Fields describing *upstream* interaction are prefixed with the string
+// `upstream_`.
+message AccessLogBaseEntry {
+  // This field indicates the rate at which this log entry was sampled.
+  // Valid range is (0.0, 1.0].
+  double sample_rate = 1;
+
+  // This field is the IP and port on which the request from the user was
+  // received, stored in ipv4:port or [ipv6]:port format.
+  string destination_host = 2;
+
+  // Whether the request arrived via a secure (TLS) protocol
+  google.protobuf.BoolValue secure = 3;
+
+  // Properties of the TLS connection, if used
+  TLSProperties tls_properties = 4;
 
   // The time that Envoy started servicing this request
-  google.protobuf.Timestamp start_time = 2;
+  google.protobuf.Timestamp start_time = 7;
 
-  // Incoming protocol variation spoken
-  enum Protocol {
-    PROTOCOL_UNSPECIFIED = 0;
-    HTTP10 = 1;
-    HTTP11 = 2;
-    HTTP2 = 3;
-  }
-  Protocol protocol_variant = 3;
+  // Duration (milliseconds)
+  //
+  // The total duration it took to service this request from the StartTime until
+  // the response was written to the user.
+  google.protobuf.Duration duration = 8;
+
+  // The upstream host URL (Envoy connects to).
+  //
+  // For example, tcp://ip:port for TCP connections.
+  //
+  // IPv6 addresses should be stored in canonical (compressed) format using
+  // [address]:port notation.
+  string upstream_host = 9;
+
+  // The Upstream Cluster that the upstream host belongs to.
+  string upstream_cluster = 10;
+
+  // Upstream Service Time Duration
+  //
+  // The is the amount of time it took the upstream server to service the request.
```

Comment:
Please be more specific.  Is this time to first byte?  Time to last byte?  Header or body? Something else?

And does this measurement make sense for TCP access logs, and if so what does it mean in that context?

Reply:
I actually debated this with myself. Ultimately I feel it could still meaningfully capture the duration of the connection, but we could also simply have `connection_duration` as a field in TCP access logs, and a some other duration field as a member in the HTTP case, to avoid conflating the two.

I very much agree with the point about being precise - I think first byte received to last byte sent would make the most sense for a total_duration field, and we might also usefully define some other intervals (last byte received to first byte sent, first byte received to last byte received, etc.).

Reply:
In the current implementation taken from the header, IIRC this is TTFB. I would actually add a field for both TTFB and TTLB from upstream. Then optimally we should also have TTFB and TTLB downstream. TTFB may be different due to filters. TTLB clearly can be different due to flow control.

Reply:
@mattklein123  I like this idea. Will add these four fields in lieu of duration. (Assuming the interval for all 4 starts at first downstream byte received.)